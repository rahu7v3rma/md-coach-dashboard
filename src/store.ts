import { configureStore } from '@reduxjs/toolkit';

import clientReducer from './reducers/client';
import groupReducer from './reducers/group';
import notificationReducer from './reducers/notification';
import userReducer from './reducers/user';

// define store options as a variable so we can export it for tests to use
const storeOptions: any = {
    reducer: {
        user: userReducer,
        client: clientReducer,
        notification: notificationReducer,
        group: groupReducer
    }
};

const store = configureStore(storeOptions);

export default store;

// infer the `RootState` type from the store
export type RootState = ReturnType<typeof store.getState>;

// infer the `AppDispatch` type from the store so thunk actions can return promises
export type AppDispatch = typeof store.dispatch;

export const test = {
    storeOptions
};
