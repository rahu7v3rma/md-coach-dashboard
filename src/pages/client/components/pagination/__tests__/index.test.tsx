import { Provider } from 'react-redux';
import { act, create } from 'react-test-renderer';

import PaginationComponent from 'src/pages/client/components/pagination';
import store from 'src/store';

const nextPageNumber = 3;
const previousPageNumber = 1;
jest.mock('src/reducers/client', () => ({
    NoteSelectors: jest.fn().mockImplementation(() => ({
        loading: true,
        nextPageNumber: nextPageNumber,
        previousPageNumber: previousPageNumber,
        hasPrevious: true,
        hasNext: true
    }))
}));

describe('PaginationComponent', () => {
    it('PaginationComponent snapshot', () => {
        const tree = create(
            <Provider store={store}>
                <PaginationComponent
                    rowCount={100}
                    rowsPerPage={10}
                    onChangePage={() => {}}
                    onChangeRowsPerPage={() => {}}
                    currentPage={1}
                />
            </Provider>
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('PaginationComponent should renders without errors', () => {
        const tree = create(
            <Provider store={store}>
                <PaginationComponent
                    rowCount={100}
                    rowsPerPage={10}
                    onChangePage={() => {}}
                    onChangeRowsPerPage={() => {}}
                    currentPage={1}
                />
            </Provider>
        );
        expect(tree).toBeTruthy();
    });

    it('renders current page and total pages correctly', () => {
        const tree = create(
            <Provider store={store}>
                <PaginationComponent
                    rowCount={100}
                    rowsPerPage={10}
                    onChangePage={() => {}}
                    onChangeRowsPerPage={() => {}}
                    currentPage={1}
                />
            </Provider>
        );

        const pageText = tree.root.findByProps({
            color: '#babfc3'
        }).props;
        expect(pageText.children).toStrictEqual([' ', 1, ' of ', 10, ' ']);
    });

    it('handles "Next" button click', async () => {
        const onChangePage = jest.fn();
        const tree = create(
            <Provider store={store}>
                <PaginationComponent
                    rowCount={100}
                    rowsPerPage={10}
                    onChangePage={onChangePage}
                    onChangeRowsPerPage={() => {}}
                    currentPage={1}
                />
            </Provider>
        );

        const nextBtn = tree.root.findByProps({
            id: 'next_btn'
        }).props;
        await act(() => nextBtn.onClick());
        expect(onChangePage).toHaveBeenCalledWith(nextPageNumber, 100);
    });

    it('handles "Previous" button click', async () => {
        const onChangePage = jest.fn();
        const tree = create(
            <Provider store={store}>
                <PaginationComponent
                    rowCount={100}
                    rowsPerPage={10}
                    onChangePage={onChangePage}
                    onChangeRowsPerPage={() => {}}
                    currentPage={2}
                />
            </Provider>
        );

        const prevBtn = tree.root.findByProps({
            id: 'prev_btn'
        }).props;
        await act(() => prevBtn.onClick());
        expect(onChangePage).toHaveBeenCalledWith(previousPageNumber, 100);
    });

    it('disables "Previous" button on the first page', async () => {
        const onChangePage = jest.fn();
        const tree = create(
            <Provider store={store}>
                <PaginationComponent
                    rowCount={100}
                    rowsPerPage={10}
                    onChangePage={onChangePage}
                    onChangeRowsPerPage={() => {}}
                    currentPage={1}
                />
            </Provider>
        );

        const prevBtn = tree.root.findByProps({
            id: 'prev_btn'
        }).props;
        await act(() => prevBtn.onClick());
        expect(onChangePage).not.toHaveBeenCalled();
    });

    it('disables "Next" button on the last page', async () => {
        const onChangePage = jest.fn();
        const tree = create(
            <Provider store={store}>
                <PaginationComponent
                    rowCount={100}
                    rowsPerPage={10}
                    onChangePage={onChangePage}
                    onChangeRowsPerPage={() => {}}
                    currentPage={10}
                />
            </Provider>
        );

        const nextBtn = tree.root.findByProps({
            id: 'next_btn'
        }).props;
        await act(() => nextBtn.onClick());
        expect(onChangePage).not.toHaveBeenCalled();
    });
});
