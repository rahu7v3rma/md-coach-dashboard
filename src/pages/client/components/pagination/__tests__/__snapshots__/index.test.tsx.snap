// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationComponent PaginationComponent snapshot 1`] = `
<div
  className="sc-gTRrQi cMOLXp"
>
  <div
    className="sc-ewnqHT bmfxqK"
  >
    <p
      className="sc-bgqQcB lckgVS"
      color="#A4AAAF"
      fontWeight="500"
    >
      Showing 
      1
       to 
      10
       of
       
      100
       Clients
    </p>
    <select
      aria-label="rows-per-page-dropwon"
      className="sc-dicizt jQqwrl"
      id="row-dropdown"
      onChange={[Function]}
      value={10}
    >
      <option
        value={10}
      >
        10
      </option>
      <option
        value={20}
      >
        20
      </option>
    </select>
  </div>
  <div
    className="sc-ewnqHT bmfxqK"
  >
    <button
      className="sc-fFGjHI gXKLjv"
      id="prev_btn"
      onClick={[Function]}
    >
      Previous
    </button>
    <p
      className="sc-bgqQcB jJSqi"
      color="#babfc3"
    >
       
      1
       of 
      10
       
    </p>
    <button
      className="sc-fFGjHI gXKLjv"
      id="next_btn"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
